### PR TITLE
Performance to detach RealmList

### PR DESCRIPTION
**Pull request by lucasdornelasv**
I created a new constructor to start with initialCapacity for gain more performance when to detach from realm.

public RealmList(int initialCapacity) {
        realm = null;
        osListOperator = null;
        unmanagedList = new ArrayList<>(initialCapacity);
    }

### CODE REVIEW

**Comment by lucasdornelasv** [comment]
Oh, Thank you @cmelchior , I'll do this as soon as possible.

File: realm/realm-library/src/main/java/io/realm/RealmList.java [hunk 1]
```diff
@@ -88,6 +88,21 @@ public RealmList() {
         unmanagedList = new ArrayList<>();
     }
 
+    /**
+     * Creates a RealmList in unmanaged mode with a initial capacity, where the elements are not controlled by a Realm.
+     * This effectively makes the RealmList function as a {@link java.util.ArrayList} and it is not possible to query
+     * the objects in this state.
+     * <p>
+     * Use {@link io.realm.Realm#copyToRealm(Iterable)} to properly persist its elements in Realm.
```
Comment (nhachicha) [inline realm/realm-library/src/main/java/io/realm/RealmList.java]: `... or {@link io.realm.Realm#insertOrUpdate(Collection)}`

File: realm/realm-library/src/main/java/io/realm/RealmList.java [hunk 2]
```diff
@@ -108,6 +123,25 @@ public RealmList(E... objects) {
         Collections.addAll(unmanagedList, objects);
     }
 
+    /**
+     * Creates a RealmList in unmanaged mode with an initial list of elements.
+     * A RealmList in unmanaged mode function as a {@link java.util.ArrayList} and it is not possible to query the
+     * objects in this state.
+     * <p>
+     * Use {@link io.realm.Realm#copyToRealm(Iterable)} to properly persist all unmanaged elements in Realm.
```
Comment (nhachicha) [inline realm/realm-library/src/main/java/io/realm/RealmList.java]: ... or {@link io.realm.Realm#insertOrUpdate(Collection)}

**Comment by bmunkholm** [comment]
@lucasdornelasv Do you intend to continue on this one?

**Comment by cmelchior** [comment]
Closing due to inactivity

**Comment by Zhuinden** [comment]
That's a shame, because that's actually a significant performance increase.

**Pull request closed without merging**